{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from synorchestrator import orchestrator\n",
    "from synorchestrator import config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Orchestrator options:\n",
      "\n",
      "Evaluation Queues\n",
      "---------------------------------------------------------------------------\n",
      "wflow0: github.com/dockstore-testing/md5sum-checker [CWL]\n",
      "wflow1: github.com/dockstore-testing/md5sum-checker/wdl [WDL]\n",
      "wflow2: github.com/DataBiosphere/topmed-workflows/TopMed_Variant_Caller [WDL]\n",
      "wflow3: github.com/DataBiosphere/topmed-workflows/u_of_Michigan_alignment_pipeline [WDL]\n",
      "\n",
      "Tool Registries\n",
      "---------------------------------------------------------------------------\n",
      "dockstore: dockstore.org:8443\n",
      "\n",
      "Workflow Services\n",
      "---------------------------------------------------------------------------\n",
      "hca-cromwell: g0n2qjnu94.execute-api.us-east-1.amazonaws.com/test\n",
      "broad-cromwell: 35.226.102.121:9090\n",
      "arvados-wes: wes.qr1hi.arvadosapi.com\n",
      "local: 0.0.0.0:8080\n"
     ]
    }
   ],
   "source": [
    "config.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:synorchestrator.orchestrator:Preparing checker workflow run request for 'github.com/DataBiosphere/topmed-workflows/u_of_Michigan_alignment_pipeline' from  'dockstore''\n",
      "INFO:synorchestrator.eval:Created new job submission:\n",
      " - evaluation queue: wflow3 (github.com/DataBiosphere/topmed-workflows/u_of_Michigan_alignment_pipeline)\n",
      " - submission ID: '290529211034671585'\n",
      " - data:\n",
      "{\n",
      "  \"workflow_descriptor\": \"import \\\"https://raw.githubusercontent.com/DataBiosphere/topmed-workflows/1.13.0/aligner/u_of_michigan_aligner/u_of_michigan_aligner.wdl\\\" as TopMed_aligner\\nimport \\\"https://raw.githubusercontent.com/DataBiosphere/topmed-workflows/1.13.0/aligner/functional-equivalence-checker/topmed-alignment-checker.wdl\\\" as checker\\n\\nworkflow checkerWorkflow {\\n  Int expectedNumofReads\\n  String docker_image\\n\\n  File input_crai_file\\n  File input_cram_file\\n\\n  File ref_alt\\n  File ref_bwt\\n  File ref_sa\\n  File ref_amb\\n  File ref_ann\\n  File ref_pac\\n\\n  File ref_fasta\\n  File ref_fasta_index\\n\\n  File dbSNP_vcf\\n  File dbSNP_vcf_index\\n\\n\\n call TopMed_aligner.TopMedAligner as aligner { \\n   input:\\n\\n     input_crai_file = input_crai_file,\\n     input_cram_file = input_cram_file,\\n     \\n     docker_image = docker_image,\\n\\n     ref_fasta = ref_fasta,\\n     ref_fasta_index = ref_fasta_index,\\n     ref_alt = ref_alt,\\n     ref_bwt = ref_bwt,\\n     ref_sa = ref_sa,\\n     ref_amb = ref_amb,\\n     ref_ann = ref_ann,\\n     ref_pac = ref_pac,\\n\\n     dbSNP_vcf = dbSNP_vcf,\\n     dbSNP_vcf_index = dbSNP_vcf_index\\n\\n }\\n\\n\\n call checker.checkerTask { input: inputCRAMFile=aligner.aligner_output, expectedNumofReads=expectedNumofReads, docker_image=docker_image }\\n}\\n\", \n",
      "  \"workflow_params\": {\n",
      "    \"checkerWorkflow.expectedNumofReads\": 352552, \n",
      "    \"checkerWorkflow.input_cram_file\": \"gs://topmed_workflow_testing/topmed_aligner/input_files/NWD176325.0005.recab.cram\", \n",
      "    \"checkerWorkflow.input_crai_file\": \"gs://topmed_workflow_testing/topmed_aligner/input_files/NWD176325.0005.recab.cram.crai\", \n",
      "    \"checkerWorkflow.ref_alt\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.alt\", \n",
      "    \"checkerWorkflow.ref_bwt\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.bwt\", \n",
      "    \"checkerWorkflow.ref_pac\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.pac\", \n",
      "    \"checkerWorkflow.ref_ann\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.ann\", \n",
      "    \"checkerWorkflow.ref_amb\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.amb\", \n",
      "    \"checkerWorkflow.ref_sa\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.sa\", \n",
      "    \"checkerWorkflow.ref_fasta\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa\", \n",
      "    \"checkerWorkflow.ref_fasta_index\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.fai\", \n",
      "    \"checkerWorkflow.dbSNP_vcf\": \"gs://topmed_workflow_testing/topmed_aligner/reference_files/hg38/Homo_sapiens_assembly38.dbsnp138.vcf.gz\", \n",
      "    \"checkerWorkflow.dbSNP_vcf_index\": \"gs://topmed_workflow_testing/topmed_aligner/reference_files/hg38/Homo_sapiens_assembly38.dbsnp138.vcf.gz.tbi\", \n",
      "    \"checkerWorkflow.docker_image\": \"statgen/alignment\"\n",
      "  }, \n",
      "  \"workflow_type_version\": \"draft-2\", \n",
      "  \"workflow_type\": \"WDL\"\n",
      "}\n",
      "INFO:synorchestrator.orchestrator:Preparing checker workflow run request for 'github.com/DataBiosphere/topmed-workflows/u_of_Michigan_alignment_pipeline' from  'dockstore''\n",
      "INFO:synorchestrator.eval:Created new job submission:\n",
      " - evaluation queue: wflow3 (github.com/DataBiosphere/topmed-workflows/u_of_Michigan_alignment_pipeline)\n",
      " - submission ID: '290529211036592693'\n",
      " - data:\n",
      "{\n",
      "  \"workflow_descriptor\": \"import \\\"https://raw.githubusercontent.com/DataBiosphere/topmed-workflows/1.13.0/aligner/u_of_michigan_aligner/u_of_michigan_aligner.wdl\\\" as TopMed_aligner\\nimport \\\"https://raw.githubusercontent.com/DataBiosphere/topmed-workflows/1.13.0/aligner/functional-equivalence-checker/topmed-alignment-checker.wdl\\\" as checker\\n\\nworkflow checkerWorkflow {\\n  Int expectedNumofReads\\n  String docker_image\\n\\n  File input_crai_file\\n  File input_cram_file\\n\\n  File ref_alt\\n  File ref_bwt\\n  File ref_sa\\n  File ref_amb\\n  File ref_ann\\n  File ref_pac\\n\\n  File ref_fasta\\n  File ref_fasta_index\\n\\n  File dbSNP_vcf\\n  File dbSNP_vcf_index\\n\\n\\n call TopMed_aligner.TopMedAligner as aligner { \\n   input:\\n\\n     input_crai_file = input_crai_file,\\n     input_cram_file = input_cram_file,\\n     \\n     docker_image = docker_image,\\n\\n     ref_fasta = ref_fasta,\\n     ref_fasta_index = ref_fasta_index,\\n     ref_alt = ref_alt,\\n     ref_bwt = ref_bwt,\\n     ref_sa = ref_sa,\\n     ref_amb = ref_amb,\\n     ref_ann = ref_ann,\\n     ref_pac = ref_pac,\\n\\n     dbSNP_vcf = dbSNP_vcf,\\n     dbSNP_vcf_index = dbSNP_vcf_index\\n\\n }\\n\\n\\n call checker.checkerTask { input: inputCRAMFile=aligner.aligner_output, expectedNumofReads=expectedNumofReads, docker_image=docker_image }\\n}\\n\", \n",
      "  \"workflow_params\": {\n",
      "    \"checkerWorkflow.expectedNumofReads\": 352552, \n",
      "    \"checkerWorkflow.input_cram_file\": \"gs://topmed_workflow_testing/topmed_aligner/input_files/NWD176325.0005.recab.cram\", \n",
      "    \"checkerWorkflow.input_crai_file\": \"gs://topmed_workflow_testing/topmed_aligner/input_files/NWD176325.0005.recab.cram.crai\", \n",
      "    \"checkerWorkflow.ref_alt\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.alt\", \n",
      "    \"checkerWorkflow.ref_bwt\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.bwt\", \n",
      "    \"checkerWorkflow.ref_pac\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.pac\", \n",
      "    \"checkerWorkflow.ref_ann\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.ann\", \n",
      "    \"checkerWorkflow.ref_amb\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.amb\", \n",
      "    \"checkerWorkflow.ref_sa\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.sa\", \n",
      "    \"checkerWorkflow.ref_fasta\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa\", \n",
      "    \"checkerWorkflow.ref_fasta_index\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.fai\", \n",
      "    \"checkerWorkflow.dbSNP_vcf\": \"gs://topmed_workflow_testing/topmed_aligner/reference_files/hg38/Homo_sapiens_assembly38.dbsnp138.vcf.gz\", \n",
      "    \"checkerWorkflow.dbSNP_vcf_index\": \"gs://topmed_workflow_testing/topmed_aligner/reference_files/hg38/Homo_sapiens_assembly38.dbsnp138.vcf.gz.tbi\", \n",
      "    \"checkerWorkflow.docker_image\": \"statgen/alignment\"\n",
      "  }, \n",
      "  \"workflow_type_version\": \"draft-2\", \n",
      "  \"workflow_type\": \"WDL\"\n",
      "}\n",
      "INFO:synorchestrator.orchestrator:Preparing checker workflow run request for 'github.com/DataBiosphere/topmed-workflows/TopMed_Variant_Caller' from  'dockstore''\n",
      "INFO:synorchestrator.eval:Created new job submission:\n",
      " - evaluation queue: wflow2 (github.com/DataBiosphere/topmed-workflows/TopMed_Variant_Caller)\n",
      " - submission ID: '290529211038681861'\n",
      " - data:\n",
      "{\n",
      "  \"workflow_descriptor\": \"import \\\"https://raw.githubusercontent.com/DataBiosphere/topmed-workflows/1.11.0/variant-caller/variant-caller-wdl/topmed_freeze3_calling.wdl\\\" as TopMed_variantcaller\\nimport \\\"https://raw.githubusercontent.com/DataBiosphere/topmed-workflows/1.11.0/variant-caller/variant-caller-wdl-checker/topmed-variantcaller-checker.wdl\\\" as checker\\n\\nworkflow checkerWorkflow {\\n  File inputTruthVCFFile\\n\\n  # Deprecated: No need to input this anymore\\n  # Disk size requirements will be calculated internally\\n  # This will be removed in the next release\\n  Float? reference_files_size\\n\\n  String docker_image\\n\\n  Array[File] input_crai_files\\n  Array[File] input_cram_files\\n\\n  File ref_1000G_omni2_5_b38_sites_PASS_vcf_gz\\n  File ref_1000G_omni2_5_b38_sites_PASS_vcf_gz_tbi\\n  File chr10_vcf\\n  File chr11_KI270927v1_alt_vcf\\n  File chr11_vcf\\n  File chr12_vcf\\n  File chr13_vcf\\n  File chr14_GL000009v2_random_vcf\\n  File chr14_KI270846v1_alt_vcf\\n  File chr14_vcf\\n  File chr15_vcf\\n  File chr16_vcf\\n  File chr17_KI270857v1_alt_vcf\\n  File chr17_KI270862v1_alt_vcf\\n  File chr17_KI270909v1_alt_vcf\\n  File chr17_vcf\\n  File chr18_vcf\\n  File chr19_KI270938v1_alt_vcf\\n  File chr19_vcf\\n  File chr1_KI270706v1_random_vcf\\n  File chr1_KI270766v1_alt_vcf\\n  File chr1_vcf\\n  File chr20_vcf\\n  File chr21_vcf\\n  File chr22_KI270879v1_alt_vcf\\n  File chr22_KI270928v1_alt_vcf\\n  File chr22_vcf\\n  File chr2_KI270773v1_alt_vcf\\n  File chr2_KI270894v1_alt_vcf\\n  File chr2_vcf\\n  File chr3_vcf\\n  File chr4_GL000008v2_random_vcf\\n  File chr4_vcf\\n  File chr5_vcf\\n  File chr6_vcf\\n  File chr7_KI270803v1_alt_vcf\\n  File chr7_vcf\\n  File chr8_KI270821v1_alt_vcf\\n  File chr8_vcf\\n  File chr9_vcf\\n  File chrUn_KI270742v1_vcf\\n  File chrX_vcf\\n  File ref_dbsnp_142_b38_vcf_gz\\n  File ref_dbsnp_142_b38_vcf_gz_tbi\\n  File ref_dbsnp_All_vcf_gz\\n  File ref_dbsnp_All_vcf_gz_tbi\\n  File ref_hapmap_3_3_b38_sites_vcf_gz\\n  File ref_hapmap_3_3_b38_sites_vcf_gz_tbi\\n  File ref_hs38DH_bs_umfa\\n  File ref_hs38DH_dict\\n  File ref_hs38DH_fa\\n  File ref_hs38DH_fa_alt\\n  File ref_hs38DH_fa_amb\\n  File ref_hs38DH_fa_ann\\n  File ref_hs38DH_fa_bwt\\n  File ref_hs38DH_fa_fai\\n  File ref_hs38DH_fa_pac\\n  File ref_hs38DH_fa_sa\\n  File ref_hs38DH_winsize100_gc\\n\\n\\n  call TopMed_variantcaller.TopMedVariantCaller as variantcaller {\\n    input:\\n      input_crai_files = input_crai_files,\\n      input_cram_files = input_cram_files,\\n\\n      docker_image = docker_image,\\n\\n      ref_1000G_omni2_5_b38_sites_PASS_vcf_gz = ref_1000G_omni2_5_b38_sites_PASS_vcf_gz,\\n      ref_1000G_omni2_5_b38_sites_PASS_vcf_gz_tbi = ref_1000G_omni2_5_b38_sites_PASS_vcf_gz_tbi,\\n      chr10_vcf = chr10_vcf,\\n      chr11_KI270927v1_alt_vcf = chr11_KI270927v1_alt_vcf,\\n      chr11_vcf = chr11_vcf,\\n      chr12_vcf = chr12_vcf,\\n      chr13_vcf = chr13_vcf,\\n      chr14_GL000009v2_random_vcf = chr14_GL000009v2_random_vcf,\\n      chr14_KI270846v1_alt_vcf = chr14_KI270846v1_alt_vcf,\\n      chr14_vcf = chr14_vcf,\\n      chr15_vcf = chr15_vcf,\\n      chr16_vcf = chr16_vcf,\\n      chr17_KI270857v1_alt_vcf = chr17_KI270857v1_alt_vcf,\\n      chr17_KI270862v1_alt_vcf = chr17_KI270862v1_alt_vcf,\\n      chr17_KI270909v1_alt_vcf = chr17_KI270909v1_alt_vcf,\\n      chr17_vcf = chr17_vcf,\\n      chr18_vcf = chr18_vcf,\\n      chr19_KI270938v1_alt_vcf = chr19_KI270938v1_alt_vcf,\\n      chr19_vcf = chr19_vcf,\\n      chr1_KI270706v1_random_vcf = chr1_KI270706v1_random_vcf,\\n      chr1_KI270766v1_alt_vcf = chr1_KI270766v1_alt_vcf,\\n      chr1_vcf = chr1_vcf,\\n      chr20_vcf = chr20_vcf,\\n      chr21_vcf = chr21_vcf,\\n      chr22_KI270879v1_alt_vcf = chr22_KI270879v1_alt_vcf,\\n      chr22_KI270928v1_alt_vcf = chr22_KI270928v1_alt_vcf,\\n      chr22_vcf = chr22_vcf,\\n      chr2_KI270773v1_alt_vcf = chr2_KI270773v1_alt_vcf,\\n      chr2_KI270894v1_alt_vcf = chr2_KI270894v1_alt_vcf,\\n      chr2_vcf = chr2_vcf,\\n      chr3_vcf = chr3_vcf,\\n      chr4_GL000008v2_random_vcf = chr4_GL000008v2_random_vcf,\\n      chr4_vcf = chr4_vcf,\\n      chr5_vcf = chr5_vcf,\\n      chr6_vcf = chr6_vcf,\\n      chr7_KI270803v1_alt_vcf = chr7_KI270803v1_alt_vcf,\\n      chr7_vcf = chr7_vcf,\\n      chr8_KI270821v1_alt_vcf = chr8_KI270821v1_alt_vcf,\\n      chr8_vcf = chr8_vcf,\\n      chr9_vcf = chr9_vcf,\\n      chrUn_KI270742v1_vcf = chrUn_KI270742v1_vcf,\\n      chrX_vcf = chrX_vcf,\\n      ref_dbsnp_142_b38_vcf_gz = ref_dbsnp_142_b38_vcf_gz,\\n      ref_dbsnp_142_b38_vcf_gz_tbi = ref_dbsnp_142_b38_vcf_gz_tbi,\\n      ref_dbsnp_All_vcf_gz = ref_dbsnp_All_vcf_gz,\\n      ref_dbsnp_All_vcf_gz_tbi = ref_dbsnp_All_vcf_gz_tbi,\\n      ref_hapmap_3_3_b38_sites_vcf_gz = ref_hapmap_3_3_b38_sites_vcf_gz,\\n      ref_hapmap_3_3_b38_sites_vcf_gz_tbi = ref_hapmap_3_3_b38_sites_vcf_gz_tbi,\\n      ref_hs38DH_bs_umfa = ref_hs38DH_bs_umfa,\\n      ref_hs38DH_dict = ref_hs38DH_dict,\\n      ref_hs38DH_fa = ref_hs38DH_fa,\\n      ref_hs38DH_fa_alt = ref_hs38DH_fa_alt,\\n      ref_hs38DH_fa_amb = ref_hs38DH_fa_amb,\\n      ref_hs38DH_fa_ann = ref_hs38DH_fa_ann,\\n      ref_hs38DH_fa_bwt = ref_hs38DH_fa_bwt,\\n      ref_hs38DH_fa_fai = ref_hs38DH_fa_fai,\\n      ref_hs38DH_fa_pac = ref_hs38DH_fa_pac,\\n      ref_hs38DH_fa_sa = ref_hs38DH_fa_sa,\\n      ref_hs38DH_winsize100_gc = ref_hs38DH_winsize100_gc\\n  }\\n\\n  call checker.checkerTask { \\n      input: \\n          inputTruthVCFFile = inputTruthVCFFile,\\n          inputTestVCFFile = variantcaller.topmed_variant_caller_output, \\n          docker_image = docker_image\\n  }\\n}\\n\\n\", \n",
      "  \"workflow_params\": {\n",
      "    \"checkerWorkflow.inputTruthVCFFile\": \"gs://topmed_workflow_testing/topmed_variant_caller_checker/truth_topmed_variant_caller_NWD176325.0005_output.tar.gz\", \n",
      "    \"checkerWorkflow.input_cram_files\": [\n",
      "      \"gs://topmed_workflow_testing/topmed_aligner/input_files/NWD176325.0005.recab.cram\"\n",
      "    ], \n",
      "    \"checkerWorkflow.input_crai_files\": [\n",
      "      \"gs://topmed_workflow_testing/topmed_aligner/input_files/NWD176325.0005.recab.cram.crai\"\n",
      "    ], \n",
      "    \"checkerWorkflow.ref_1000G_omni2_5_b38_sites_PASS_vcf_gz\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1000G_omni2.5.b38.sites.PASS.vcf.gz\", \n",
      "    \"checkerWorkflow.ref_1000G_omni2_5_b38_sites_PASS_vcf_gz_tbi\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1000G_omni2.5.b38.sites.PASS.vcf.gz.tbi\", \n",
      "    \"checkerWorkflow.chr10_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr10.vcf\", \n",
      "    \"checkerWorkflow.chr11_KI270927v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr11_KI270927v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr11_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr11.vcf\", \n",
      "    \"checkerWorkflow.chr12_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr12.vcf\", \n",
      "    \"checkerWorkflow.chr13_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr13.vcf\", \n",
      "    \"checkerWorkflow.chr14_GL000009v2_random_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr14_GL000009v2_random.vcf\", \n",
      "    \"checkerWorkflow.chr14_KI270846v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr14_KI270846v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr14_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr14.vcf\", \n",
      "    \"checkerWorkflow.chr15_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr15.vcf\", \n",
      "    \"checkerWorkflow.chr16_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr16.vcf\", \n",
      "    \"checkerWorkflow.chr17_KI270857v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr17_KI270857v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr17_KI270862v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr17_KI270862v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr17_KI270909v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr17_KI270909v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr17_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr17.vcf\", \n",
      "    \"checkerWorkflow.chr18_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr18.vcf\", \n",
      "    \"checkerWorkflow.chr19_KI270938v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr19_KI270938v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr19_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr19.vcf\", \n",
      "    \"checkerWorkflow.chr1_KI270706v1_random_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr1_KI270706v1_random.vcf\", \n",
      "    \"checkerWorkflow.chr1_KI270766v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr1_KI270766v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr1_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr1.vcf\", \n",
      "    \"checkerWorkflow.chr20_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr20.vcf\", \n",
      "    \"checkerWorkflow.chr21_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr21.vcf\", \n",
      "    \"checkerWorkflow.chr22_KI270879v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr22_KI270879v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr22_KI270928v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr22_KI270928v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr22_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr22.vcf\", \n",
      "    \"checkerWorkflow.chr2_KI270773v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr2_KI270773v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr2_KI270894v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr2_KI270894v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr2_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr2.vcf\", \n",
      "    \"checkerWorkflow.chr3_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr3.vcf\", \n",
      "    \"checkerWorkflow.chr4_GL000008v2_random_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr4_GL000008v2_random.vcf\", \n",
      "    \"checkerWorkflow.chr4_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr4.vcf\", \n",
      "    \"checkerWorkflow.chr5_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr5.vcf\", \n",
      "    \"checkerWorkflow.chr6_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr6.vcf\", \n",
      "    \"checkerWorkflow.chr7_KI270803v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr7_KI270803v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr7_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr7.vcf\", \n",
      "    \"checkerWorkflow.chr8_KI270821v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr8_KI270821v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr8_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr8.vcf\", \n",
      "    \"checkerWorkflow.chr9_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr9.vcf\", \n",
      "    \"checkerWorkflow.chrUn_KI270742v1_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chrUn_KI270742v1.vcf\", \n",
      "    \"checkerWorkflow.chrX_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chrX.vcf\", \n",
      "    \"checkerWorkflow.ref_dbsnp_142_b38_vcf_gz\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/dbsnp_142.b38.vcf.gz\", \n",
      "    \"checkerWorkflow.ref_dbsnp_142_b38_vcf_gz_tbi\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/dbsnp_142.b38.vcf.gz.tbi\", \n",
      "    \"checkerWorkflow.ref_dbsnp_All_vcf_gz\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/dbsnp.All.vcf.gz\", \n",
      "    \"checkerWorkflow.ref_dbsnp_All_vcf_gz_tbi\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/dbsnp.All.vcf.gz.tbi\", \n",
      "    \"checkerWorkflow.ref_hapmap_3_3_b38_sites_vcf_gz\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hapmap_3.3.b38.sites.vcf.gz\", \n",
      "    \"checkerWorkflow.ref_hapmap_3_3_b38_sites_vcf_gz_tbi\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hapmap_3.3.b38.sites.vcf.gz.tbi\", \n",
      "    \"checkerWorkflow.ref_hs38DH_bs_umfa\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH-bs.umfa\", \n",
      "    \"checkerWorkflow.ref_hs38DH_dict\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.dict\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa_alt\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.alt\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa_amb\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.amb\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa_ann\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.ann\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa_bwt\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.bwt\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa_fai\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.fai\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa_pac\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.pac\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa_sa\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.sa\", \n",
      "    \"checkerWorkflow.ref_hs38DH_winsize100_gc\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.winsize100.gc\", \n",
      "    \"checkerWorkflow.docker_image\": \"quay.io/ucsc_cgl/topmed_freeze3_calling:1.9.0\"\n",
      "  }, \n",
      "  \"workflow_type_version\": \"draft-2\", \n",
      "  \"workflow_type\": \"WDL\"\n",
      "}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:synorchestrator.orchestrator:Preparing checker workflow run request for 'github.com/DataBiosphere/topmed-workflows/TopMed_Variant_Caller' from  'dockstore''\n",
      "INFO:synorchestrator.eval:Created new job submission:\n",
      " - evaluation queue: wflow2 (github.com/DataBiosphere/topmed-workflows/TopMed_Variant_Caller)\n",
      " - submission ID: '290529211040766509'\n",
      " - data:\n",
      "{\n",
      "  \"workflow_descriptor\": \"import \\\"https://raw.githubusercontent.com/DataBiosphere/topmed-workflows/1.11.0/variant-caller/variant-caller-wdl/topmed_freeze3_calling.wdl\\\" as TopMed_variantcaller\\nimport \\\"https://raw.githubusercontent.com/DataBiosphere/topmed-workflows/1.11.0/variant-caller/variant-caller-wdl-checker/topmed-variantcaller-checker.wdl\\\" as checker\\n\\nworkflow checkerWorkflow {\\n  File inputTruthVCFFile\\n\\n  # Deprecated: No need to input this anymore\\n  # Disk size requirements will be calculated internally\\n  # This will be removed in the next release\\n  Float? reference_files_size\\n\\n  String docker_image\\n\\n  Array[File] input_crai_files\\n  Array[File] input_cram_files\\n\\n  File ref_1000G_omni2_5_b38_sites_PASS_vcf_gz\\n  File ref_1000G_omni2_5_b38_sites_PASS_vcf_gz_tbi\\n  File chr10_vcf\\n  File chr11_KI270927v1_alt_vcf\\n  File chr11_vcf\\n  File chr12_vcf\\n  File chr13_vcf\\n  File chr14_GL000009v2_random_vcf\\n  File chr14_KI270846v1_alt_vcf\\n  File chr14_vcf\\n  File chr15_vcf\\n  File chr16_vcf\\n  File chr17_KI270857v1_alt_vcf\\n  File chr17_KI270862v1_alt_vcf\\n  File chr17_KI270909v1_alt_vcf\\n  File chr17_vcf\\n  File chr18_vcf\\n  File chr19_KI270938v1_alt_vcf\\n  File chr19_vcf\\n  File chr1_KI270706v1_random_vcf\\n  File chr1_KI270766v1_alt_vcf\\n  File chr1_vcf\\n  File chr20_vcf\\n  File chr21_vcf\\n  File chr22_KI270879v1_alt_vcf\\n  File chr22_KI270928v1_alt_vcf\\n  File chr22_vcf\\n  File chr2_KI270773v1_alt_vcf\\n  File chr2_KI270894v1_alt_vcf\\n  File chr2_vcf\\n  File chr3_vcf\\n  File chr4_GL000008v2_random_vcf\\n  File chr4_vcf\\n  File chr5_vcf\\n  File chr6_vcf\\n  File chr7_KI270803v1_alt_vcf\\n  File chr7_vcf\\n  File chr8_KI270821v1_alt_vcf\\n  File chr8_vcf\\n  File chr9_vcf\\n  File chrUn_KI270742v1_vcf\\n  File chrX_vcf\\n  File ref_dbsnp_142_b38_vcf_gz\\n  File ref_dbsnp_142_b38_vcf_gz_tbi\\n  File ref_dbsnp_All_vcf_gz\\n  File ref_dbsnp_All_vcf_gz_tbi\\n  File ref_hapmap_3_3_b38_sites_vcf_gz\\n  File ref_hapmap_3_3_b38_sites_vcf_gz_tbi\\n  File ref_hs38DH_bs_umfa\\n  File ref_hs38DH_dict\\n  File ref_hs38DH_fa\\n  File ref_hs38DH_fa_alt\\n  File ref_hs38DH_fa_amb\\n  File ref_hs38DH_fa_ann\\n  File ref_hs38DH_fa_bwt\\n  File ref_hs38DH_fa_fai\\n  File ref_hs38DH_fa_pac\\n  File ref_hs38DH_fa_sa\\n  File ref_hs38DH_winsize100_gc\\n\\n\\n  call TopMed_variantcaller.TopMedVariantCaller as variantcaller {\\n    input:\\n      input_crai_files = input_crai_files,\\n      input_cram_files = input_cram_files,\\n\\n      docker_image = docker_image,\\n\\n      ref_1000G_omni2_5_b38_sites_PASS_vcf_gz = ref_1000G_omni2_5_b38_sites_PASS_vcf_gz,\\n      ref_1000G_omni2_5_b38_sites_PASS_vcf_gz_tbi = ref_1000G_omni2_5_b38_sites_PASS_vcf_gz_tbi,\\n      chr10_vcf = chr10_vcf,\\n      chr11_KI270927v1_alt_vcf = chr11_KI270927v1_alt_vcf,\\n      chr11_vcf = chr11_vcf,\\n      chr12_vcf = chr12_vcf,\\n      chr13_vcf = chr13_vcf,\\n      chr14_GL000009v2_random_vcf = chr14_GL000009v2_random_vcf,\\n      chr14_KI270846v1_alt_vcf = chr14_KI270846v1_alt_vcf,\\n      chr14_vcf = chr14_vcf,\\n      chr15_vcf = chr15_vcf,\\n      chr16_vcf = chr16_vcf,\\n      chr17_KI270857v1_alt_vcf = chr17_KI270857v1_alt_vcf,\\n      chr17_KI270862v1_alt_vcf = chr17_KI270862v1_alt_vcf,\\n      chr17_KI270909v1_alt_vcf = chr17_KI270909v1_alt_vcf,\\n      chr17_vcf = chr17_vcf,\\n      chr18_vcf = chr18_vcf,\\n      chr19_KI270938v1_alt_vcf = chr19_KI270938v1_alt_vcf,\\n      chr19_vcf = chr19_vcf,\\n      chr1_KI270706v1_random_vcf = chr1_KI270706v1_random_vcf,\\n      chr1_KI270766v1_alt_vcf = chr1_KI270766v1_alt_vcf,\\n      chr1_vcf = chr1_vcf,\\n      chr20_vcf = chr20_vcf,\\n      chr21_vcf = chr21_vcf,\\n      chr22_KI270879v1_alt_vcf = chr22_KI270879v1_alt_vcf,\\n      chr22_KI270928v1_alt_vcf = chr22_KI270928v1_alt_vcf,\\n      chr22_vcf = chr22_vcf,\\n      chr2_KI270773v1_alt_vcf = chr2_KI270773v1_alt_vcf,\\n      chr2_KI270894v1_alt_vcf = chr2_KI270894v1_alt_vcf,\\n      chr2_vcf = chr2_vcf,\\n      chr3_vcf = chr3_vcf,\\n      chr4_GL000008v2_random_vcf = chr4_GL000008v2_random_vcf,\\n      chr4_vcf = chr4_vcf,\\n      chr5_vcf = chr5_vcf,\\n      chr6_vcf = chr6_vcf,\\n      chr7_KI270803v1_alt_vcf = chr7_KI270803v1_alt_vcf,\\n      chr7_vcf = chr7_vcf,\\n      chr8_KI270821v1_alt_vcf = chr8_KI270821v1_alt_vcf,\\n      chr8_vcf = chr8_vcf,\\n      chr9_vcf = chr9_vcf,\\n      chrUn_KI270742v1_vcf = chrUn_KI270742v1_vcf,\\n      chrX_vcf = chrX_vcf,\\n      ref_dbsnp_142_b38_vcf_gz = ref_dbsnp_142_b38_vcf_gz,\\n      ref_dbsnp_142_b38_vcf_gz_tbi = ref_dbsnp_142_b38_vcf_gz_tbi,\\n      ref_dbsnp_All_vcf_gz = ref_dbsnp_All_vcf_gz,\\n      ref_dbsnp_All_vcf_gz_tbi = ref_dbsnp_All_vcf_gz_tbi,\\n      ref_hapmap_3_3_b38_sites_vcf_gz = ref_hapmap_3_3_b38_sites_vcf_gz,\\n      ref_hapmap_3_3_b38_sites_vcf_gz_tbi = ref_hapmap_3_3_b38_sites_vcf_gz_tbi,\\n      ref_hs38DH_bs_umfa = ref_hs38DH_bs_umfa,\\n      ref_hs38DH_dict = ref_hs38DH_dict,\\n      ref_hs38DH_fa = ref_hs38DH_fa,\\n      ref_hs38DH_fa_alt = ref_hs38DH_fa_alt,\\n      ref_hs38DH_fa_amb = ref_hs38DH_fa_amb,\\n      ref_hs38DH_fa_ann = ref_hs38DH_fa_ann,\\n      ref_hs38DH_fa_bwt = ref_hs38DH_fa_bwt,\\n      ref_hs38DH_fa_fai = ref_hs38DH_fa_fai,\\n      ref_hs38DH_fa_pac = ref_hs38DH_fa_pac,\\n      ref_hs38DH_fa_sa = ref_hs38DH_fa_sa,\\n      ref_hs38DH_winsize100_gc = ref_hs38DH_winsize100_gc\\n  }\\n\\n  call checker.checkerTask { \\n      input: \\n          inputTruthVCFFile = inputTruthVCFFile,\\n          inputTestVCFFile = variantcaller.topmed_variant_caller_output, \\n          docker_image = docker_image\\n  }\\n}\\n\\n\", \n",
      "  \"workflow_params\": {\n",
      "    \"checkerWorkflow.inputTruthVCFFile\": \"gs://topmed_workflow_testing/topmed_variant_caller_checker/truth_topmed_variant_caller_NWD176325.0005_output.tar.gz\", \n",
      "    \"checkerWorkflow.input_cram_files\": [\n",
      "      \"gs://topmed_workflow_testing/topmed_aligner/input_files/NWD176325.0005.recab.cram\"\n",
      "    ], \n",
      "    \"checkerWorkflow.input_crai_files\": [\n",
      "      \"gs://topmed_workflow_testing/topmed_aligner/input_files/NWD176325.0005.recab.cram.crai\"\n",
      "    ], \n",
      "    \"checkerWorkflow.ref_1000G_omni2_5_b38_sites_PASS_vcf_gz\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1000G_omni2.5.b38.sites.PASS.vcf.gz\", \n",
      "    \"checkerWorkflow.ref_1000G_omni2_5_b38_sites_PASS_vcf_gz_tbi\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1000G_omni2.5.b38.sites.PASS.vcf.gz.tbi\", \n",
      "    \"checkerWorkflow.chr10_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr10.vcf\", \n",
      "    \"checkerWorkflow.chr11_KI270927v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr11_KI270927v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr11_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr11.vcf\", \n",
      "    \"checkerWorkflow.chr12_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr12.vcf\", \n",
      "    \"checkerWorkflow.chr13_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr13.vcf\", \n",
      "    \"checkerWorkflow.chr14_GL000009v2_random_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr14_GL000009v2_random.vcf\", \n",
      "    \"checkerWorkflow.chr14_KI270846v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr14_KI270846v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr14_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr14.vcf\", \n",
      "    \"checkerWorkflow.chr15_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr15.vcf\", \n",
      "    \"checkerWorkflow.chr16_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr16.vcf\", \n",
      "    \"checkerWorkflow.chr17_KI270857v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr17_KI270857v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr17_KI270862v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr17_KI270862v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr17_KI270909v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr17_KI270909v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr17_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr17.vcf\", \n",
      "    \"checkerWorkflow.chr18_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr18.vcf\", \n",
      "    \"checkerWorkflow.chr19_KI270938v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr19_KI270938v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr19_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr19.vcf\", \n",
      "    \"checkerWorkflow.chr1_KI270706v1_random_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr1_KI270706v1_random.vcf\", \n",
      "    \"checkerWorkflow.chr1_KI270766v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr1_KI270766v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr1_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr1.vcf\", \n",
      "    \"checkerWorkflow.chr20_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr20.vcf\", \n",
      "    \"checkerWorkflow.chr21_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr21.vcf\", \n",
      "    \"checkerWorkflow.chr22_KI270879v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr22_KI270879v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr22_KI270928v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr22_KI270928v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr22_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr22.vcf\", \n",
      "    \"checkerWorkflow.chr2_KI270773v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr2_KI270773v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr2_KI270894v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr2_KI270894v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr2_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr2.vcf\", \n",
      "    \"checkerWorkflow.chr3_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr3.vcf\", \n",
      "    \"checkerWorkflow.chr4_GL000008v2_random_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr4_GL000008v2_random.vcf\", \n",
      "    \"checkerWorkflow.chr4_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr4.vcf\", \n",
      "    \"checkerWorkflow.chr5_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr5.vcf\", \n",
      "    \"checkerWorkflow.chr6_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr6.vcf\", \n",
      "    \"checkerWorkflow.chr7_KI270803v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr7_KI270803v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr7_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr7.vcf\", \n",
      "    \"checkerWorkflow.chr8_KI270821v1_alt_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr8_KI270821v1_alt.vcf\", \n",
      "    \"checkerWorkflow.chr8_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr8.vcf\", \n",
      "    \"checkerWorkflow.chr9_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chr9.vcf\", \n",
      "    \"checkerWorkflow.chrUn_KI270742v1_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chrUn_KI270742v1.vcf\", \n",
      "    \"checkerWorkflow.chrX_vcf\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/1kg.pilot_release.merged.indels.sites.hg38.chrX.vcf\", \n",
      "    \"checkerWorkflow.ref_dbsnp_142_b38_vcf_gz\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/dbsnp_142.b38.vcf.gz\", \n",
      "    \"checkerWorkflow.ref_dbsnp_142_b38_vcf_gz_tbi\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/dbsnp_142.b38.vcf.gz.tbi\", \n",
      "    \"checkerWorkflow.ref_dbsnp_All_vcf_gz\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/dbsnp.All.vcf.gz\", \n",
      "    \"checkerWorkflow.ref_dbsnp_All_vcf_gz_tbi\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/dbsnp.All.vcf.gz.tbi\", \n",
      "    \"checkerWorkflow.ref_hapmap_3_3_b38_sites_vcf_gz\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hapmap_3.3.b38.sites.vcf.gz\", \n",
      "    \"checkerWorkflow.ref_hapmap_3_3_b38_sites_vcf_gz_tbi\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hapmap_3.3.b38.sites.vcf.gz.tbi\", \n",
      "    \"checkerWorkflow.ref_hs38DH_bs_umfa\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH-bs.umfa\", \n",
      "    \"checkerWorkflow.ref_hs38DH_dict\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.dict\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa_alt\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.alt\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa_amb\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.amb\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa_ann\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.ann\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa_bwt\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.bwt\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa_fai\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.fai\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa_pac\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.pac\", \n",
      "    \"checkerWorkflow.ref_hs38DH_fa_sa\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.fa.sa\", \n",
      "    \"checkerWorkflow.ref_hs38DH_winsize100_gc\": \"gs://topmed_workflow_testing/topmed_variant_caller/reference_files/hg38/hs38DH.winsize100.gc\", \n",
      "    \"checkerWorkflow.docker_image\": \"quay.io/ucsc_cgl/topmed_freeze3_calling:1.9.0\"\n",
      "  }, \n",
      "  \"workflow_type_version\": \"draft-2\", \n",
      "  \"workflow_type\": \"WDL\"\n",
      "}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:synorchestrator.orchestrator:Preparing checker workflow run request for 'github.com/dockstore-testing/md5sum-checker' from  'dockstore''\n",
      "INFO:synorchestrator.eval:Created new job submission:\n",
      " - evaluation queue: wflow0 (github.com/dockstore-testing/md5sum-checker)\n",
      " - submission ID: '290529211044953971'\n",
      " - data:\n",
      "{\n",
      "  \"workflow_descriptor\": \"{\\n    \\\"cwlVersion\\\": \\\"v1.0\\\", \\n    \\\"$graph\\\": [\\n        {\\n            \\\"class\\\": \\\"Workflow\\\", \\n            \\\"requirements\\\": [\\n                {\\n                    \\\"class\\\": \\\"SubworkflowFeatureRequirement\\\"\\n                }\\n            ], \\n            \\\"inputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"string\\\", \\n                    \\\"id\\\": \\\"#main/expected_md5\\\"\\n                }, \\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"id\\\": \\\"#main/input_file\\\"\\n                }\\n            ], \\n            \\\"outputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"outputSource\\\": \\\"#main/checker/results_file\\\", \\n                    \\\"id\\\": \\\"#main/workflow_output_file\\\"\\n                }\\n            ], \\n            \\\"steps\\\": [\\n                {\\n                    \\\"run\\\": \\\"#md5sum-checker.cwl\\\", \\n                    \\\"in\\\": [\\n                        {\\n                            \\\"source\\\": \\\"#main/expected_md5\\\", \\n                            \\\"id\\\": \\\"#main/checker/expected_md5\\\"\\n                        }, \\n                        {\\n                            \\\"source\\\": \\\"#main/md5sum/output_file\\\", \\n                            \\\"id\\\": \\\"#main/checker/input_file\\\"\\n                        }\\n                    ], \\n                    \\\"out\\\": [\\n                        \\\"#main/checker/results_file\\\"\\n                    ], \\n                    \\\"id\\\": \\\"#main/checker\\\"\\n                }, \\n                {\\n                    \\\"run\\\": \\\"#md5sum-workflow.cwl\\\", \\n                    \\\"in\\\": [\\n                        {\\n                            \\\"source\\\": \\\"#main/input_file\\\", \\n                            \\\"id\\\": \\\"#main/md5sum/input_file\\\"\\n                        }\\n                    ], \\n                    \\\"out\\\": [\\n                        \\\"#main/md5sum/output_file\\\"\\n                    ], \\n                    \\\"id\\\": \\\"#main/md5sum\\\"\\n                }\\n            ], \\n            \\\"doc\\\": \\\"This demonstrates how to wrap a \\\\\\\"real\\\\\\\" tool with a checker workflow that runs both the tool and a tool that performs verification of results\\\\n\\\", \\n            \\\"id\\\": \\\"#main\\\"\\n        }, \\n        {\\n            \\\"class\\\": \\\"CommandLineTool\\\", \\n            \\\"id\\\": \\\"#md5sum-checker.cwl\\\", \\n            \\\"label\\\": \\\"A tool that checks the md5sum workflow\\\", \\n            \\\"requirements\\\": [\\n                {\\n                    \\\"class\\\": \\\"DockerRequirement\\\", \\n                    \\\"dockerPull\\\": \\\"quay.io/agduncan94/checker-md5sum\\\"\\n                }, \\n                {\\n                    \\\"class\\\": \\\"InlineJavascriptRequirement\\\"\\n                }\\n            ], \\n            \\\"hints\\\": [\\n                {\\n                    \\\"class\\\": \\\"ResourceRequirement\\\", \\n                    \\\"coresMin\\\": 1, \\n                    \\\"ramMin\\\": 1024, \\n                    \\\"outdirMin\\\": 512000\\n                }\\n            ], \\n            \\\"inputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"string\\\", \\n                    \\\"inputBinding\\\": {\\n                        \\\"position\\\": 2, \\n                        \\\"prefix\\\": \\\"--md5\\\"\\n                    }, \\n                    \\\"id\\\": \\\"#md5sum-checker.cwl/expected_md5\\\"\\n                }, \\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"inputBinding\\\": {\\n                        \\\"position\\\": 1, \\n                        \\\"prefix\\\": \\\"--input-file\\\"\\n                    }, \\n                    \\\"doc\\\": \\\"The file that contains an md5sum.\\\", \\n                    \\\"id\\\": \\\"#md5sum-checker.cwl/input_file\\\"\\n                }\\n            ], \\n            \\\"outputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"outputBinding\\\": {\\n                        \\\"glob\\\": \\\"log.txt\\\"\\n                    }, \\n                    \\\"doc\\\": \\\"A text log file that contains more details.\\\", \\n                    \\\"id\\\": \\\"#md5sum-checker.cwl/log_file\\\"\\n                }, \\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"outputBinding\\\": {\\n                        \\\"glob\\\": \\\"results.json\\\"\\n                    }, \\n                    \\\"doc\\\": \\\"A json file that contains the result of the test.\\\", \\n                    \\\"id\\\": \\\"#md5sum-checker.cwl/results_file\\\"\\n                }\\n            ], \\n            \\\"baseCommand\\\": [\\n                \\\"/bin/check_md5sum\\\"\\n            ]\\n        }, \\n        {\\n            \\\"class\\\": \\\"CommandLineTool\\\", \\n            \\\"id\\\": \\\"#md5sum-tool.cwl\\\", \\n            \\\"label\\\": \\\"Simple md5sum tool\\\", \\n            \\\"requirements\\\": [\\n                {\\n                    \\\"class\\\": \\\"DockerRequirement\\\", \\n                    \\\"dockerPull\\\": \\\"quay.io/agduncan94/my-md5sum\\\"\\n                }, \\n                {\\n                    \\\"class\\\": \\\"InlineJavascriptRequirement\\\"\\n                }\\n            ], \\n            \\\"hints\\\": [\\n                {\\n                    \\\"class\\\": \\\"ResourceRequirement\\\", \\n                    \\\"coresMin\\\": 1, \\n                    \\\"ramMin\\\": 1024, \\n                    \\\"outdirMin\\\": 512\\n                }\\n            ], \\n            \\\"inputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"inputBinding\\\": {\\n                        \\\"position\\\": 1\\n                    }, \\n                    \\\"doc\\\": \\\"The file that will have its md5sum calculated.\\\", \\n                    \\\"id\\\": \\\"#md5sum-tool.cwl/input_file\\\"\\n                }\\n            ], \\n            \\\"outputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"format\\\": \\\"http://edamontology.org/data_3671\\\", \\n                    \\\"outputBinding\\\": {\\n                        \\\"glob\\\": \\\"md5sum.txt\\\"\\n                    }, \\n                    \\\"doc\\\": \\\"A text file that contains a single line that is the md5sum of the input file.\\\", \\n                    \\\"id\\\": \\\"#md5sum-tool.cwl/output_file\\\"\\n                }\\n            ], \\n            \\\"baseCommand\\\": [\\n                \\\"/bin/my_md5sum\\\"\\n            ]\\n        }, \\n        {\\n            \\\"class\\\": \\\"Workflow\\\", \\n            \\\"inputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"id\\\": \\\"#md5sum-workflow.cwl/input_file\\\"\\n                }\\n            ], \\n            \\\"outputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"outputSource\\\": \\\"#md5sum-workflow.cwl/md5sum/output_file\\\", \\n                    \\\"id\\\": \\\"#md5sum-workflow.cwl/output_file\\\"\\n                }\\n            ], \\n            \\\"steps\\\": [\\n                {\\n                    \\\"run\\\": \\\"#md5sum-tool.cwl\\\", \\n                    \\\"in\\\": [\\n                        {\\n                            \\\"source\\\": \\\"#md5sum-workflow.cwl/input_file\\\", \\n                            \\\"id\\\": \\\"#md5sum-workflow.cwl/md5sum/input_file\\\"\\n                        }\\n                    ], \\n                    \\\"out\\\": [\\n                        \\\"#md5sum-workflow.cwl/md5sum/output_file\\\"\\n                    ], \\n                    \\\"id\\\": \\\"#md5sum-workflow.cwl/md5sum\\\"\\n                }\\n            ], \\n            \\\"id\\\": \\\"#md5sum-workflow.cwl\\\"\\n        }\\n    ]\\n}\", \n",
      "  \"workflow_params\": {\n",
      "    \"input_file\": {\n",
      "      \"class\": \"File\", \n",
      "      \"location\": \"https://raw.githubusercontent.com/dockstore-testing/md5sum-checker/develop/md5sum.input\"\n",
      "    }, \n",
      "    \"expected_md5\": \"00579a00e3e7fa0674428ac7049423e2\"\n",
      "  }, \n",
      "  \"workflow_type_version\": \"v1.0\", \n",
      "  \"workflow_type\": \"CWL\"\n",
      "}\n",
      "INFO:synorchestrator.orchestrator:Preparing checker workflow run request for 'github.com/dockstore-testing/md5sum-checker' from  'dockstore''\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:synorchestrator.eval:Created new job submission:\n",
      " - evaluation queue: wflow0 (github.com/dockstore-testing/md5sum-checker)\n",
      " - submission ID: '290529211048309644'\n",
      " - data:\n",
      "{\n",
      "  \"workflow_descriptor\": \"{\\n    \\\"cwlVersion\\\": \\\"v1.0\\\", \\n    \\\"$graph\\\": [\\n        {\\n            \\\"class\\\": \\\"Workflow\\\", \\n            \\\"requirements\\\": [\\n                {\\n                    \\\"class\\\": \\\"SubworkflowFeatureRequirement\\\"\\n                }\\n            ], \\n            \\\"inputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"string\\\", \\n                    \\\"id\\\": \\\"#main/expected_md5\\\"\\n                }, \\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"id\\\": \\\"#main/input_file\\\"\\n                }\\n            ], \\n            \\\"outputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"outputSource\\\": \\\"#main/checker/results_file\\\", \\n                    \\\"id\\\": \\\"#main/workflow_output_file\\\"\\n                }\\n            ], \\n            \\\"steps\\\": [\\n                {\\n                    \\\"run\\\": \\\"#md5sum-checker.cwl\\\", \\n                    \\\"in\\\": [\\n                        {\\n                            \\\"source\\\": \\\"#main/expected_md5\\\", \\n                            \\\"id\\\": \\\"#main/checker/expected_md5\\\"\\n                        }, \\n                        {\\n                            \\\"source\\\": \\\"#main/md5sum/output_file\\\", \\n                            \\\"id\\\": \\\"#main/checker/input_file\\\"\\n                        }\\n                    ], \\n                    \\\"out\\\": [\\n                        \\\"#main/checker/results_file\\\"\\n                    ], \\n                    \\\"id\\\": \\\"#main/checker\\\"\\n                }, \\n                {\\n                    \\\"run\\\": \\\"#md5sum-workflow.cwl\\\", \\n                    \\\"in\\\": [\\n                        {\\n                            \\\"source\\\": \\\"#main/input_file\\\", \\n                            \\\"id\\\": \\\"#main/md5sum/input_file\\\"\\n                        }\\n                    ], \\n                    \\\"out\\\": [\\n                        \\\"#main/md5sum/output_file\\\"\\n                    ], \\n                    \\\"id\\\": \\\"#main/md5sum\\\"\\n                }\\n            ], \\n            \\\"doc\\\": \\\"This demonstrates how to wrap a \\\\\\\"real\\\\\\\" tool with a checker workflow that runs both the tool and a tool that performs verification of results\\\\n\\\", \\n            \\\"id\\\": \\\"#main\\\"\\n        }, \\n        {\\n            \\\"class\\\": \\\"CommandLineTool\\\", \\n            \\\"id\\\": \\\"#md5sum-checker.cwl\\\", \\n            \\\"label\\\": \\\"A tool that checks the md5sum workflow\\\", \\n            \\\"requirements\\\": [\\n                {\\n                    \\\"class\\\": \\\"DockerRequirement\\\", \\n                    \\\"dockerPull\\\": \\\"quay.io/agduncan94/checker-md5sum\\\"\\n                }, \\n                {\\n                    \\\"class\\\": \\\"InlineJavascriptRequirement\\\"\\n                }\\n            ], \\n            \\\"hints\\\": [\\n                {\\n                    \\\"class\\\": \\\"ResourceRequirement\\\", \\n                    \\\"coresMin\\\": 1, \\n                    \\\"ramMin\\\": 1024, \\n                    \\\"outdirMin\\\": 512000\\n                }\\n            ], \\n            \\\"inputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"string\\\", \\n                    \\\"inputBinding\\\": {\\n                        \\\"position\\\": 2, \\n                        \\\"prefix\\\": \\\"--md5\\\"\\n                    }, \\n                    \\\"id\\\": \\\"#md5sum-checker.cwl/expected_md5\\\"\\n                }, \\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"inputBinding\\\": {\\n                        \\\"position\\\": 1, \\n                        \\\"prefix\\\": \\\"--input-file\\\"\\n                    }, \\n                    \\\"doc\\\": \\\"The file that contains an md5sum.\\\", \\n                    \\\"id\\\": \\\"#md5sum-checker.cwl/input_file\\\"\\n                }\\n            ], \\n            \\\"outputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"outputBinding\\\": {\\n                        \\\"glob\\\": \\\"log.txt\\\"\\n                    }, \\n                    \\\"doc\\\": \\\"A text log file that contains more details.\\\", \\n                    \\\"id\\\": \\\"#md5sum-checker.cwl/log_file\\\"\\n                }, \\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"outputBinding\\\": {\\n                        \\\"glob\\\": \\\"results.json\\\"\\n                    }, \\n                    \\\"doc\\\": \\\"A json file that contains the result of the test.\\\", \\n                    \\\"id\\\": \\\"#md5sum-checker.cwl/results_file\\\"\\n                }\\n            ], \\n            \\\"baseCommand\\\": [\\n                \\\"/bin/check_md5sum\\\"\\n            ]\\n        }, \\n        {\\n            \\\"class\\\": \\\"CommandLineTool\\\", \\n            \\\"id\\\": \\\"#md5sum-tool.cwl\\\", \\n            \\\"label\\\": \\\"Simple md5sum tool\\\", \\n            \\\"requirements\\\": [\\n                {\\n                    \\\"class\\\": \\\"DockerRequirement\\\", \\n                    \\\"dockerPull\\\": \\\"quay.io/agduncan94/my-md5sum\\\"\\n                }, \\n                {\\n                    \\\"class\\\": \\\"InlineJavascriptRequirement\\\"\\n                }\\n            ], \\n            \\\"hints\\\": [\\n                {\\n                    \\\"class\\\": \\\"ResourceRequirement\\\", \\n                    \\\"coresMin\\\": 1, \\n                    \\\"ramMin\\\": 1024, \\n                    \\\"outdirMin\\\": 512\\n                }\\n            ], \\n            \\\"inputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"inputBinding\\\": {\\n                        \\\"position\\\": 1\\n                    }, \\n                    \\\"doc\\\": \\\"The file that will have its md5sum calculated.\\\", \\n                    \\\"id\\\": \\\"#md5sum-tool.cwl/input_file\\\"\\n                }\\n            ], \\n            \\\"outputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"format\\\": \\\"http://edamontology.org/data_3671\\\", \\n                    \\\"outputBinding\\\": {\\n                        \\\"glob\\\": \\\"md5sum.txt\\\"\\n                    }, \\n                    \\\"doc\\\": \\\"A text file that contains a single line that is the md5sum of the input file.\\\", \\n                    \\\"id\\\": \\\"#md5sum-tool.cwl/output_file\\\"\\n                }\\n            ], \\n            \\\"baseCommand\\\": [\\n                \\\"/bin/my_md5sum\\\"\\n            ]\\n        }, \\n        {\\n            \\\"class\\\": \\\"Workflow\\\", \\n            \\\"inputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"id\\\": \\\"#md5sum-workflow.cwl/input_file\\\"\\n                }\\n            ], \\n            \\\"outputs\\\": [\\n                {\\n                    \\\"type\\\": \\\"File\\\", \\n                    \\\"outputSource\\\": \\\"#md5sum-workflow.cwl/md5sum/output_file\\\", \\n                    \\\"id\\\": \\\"#md5sum-workflow.cwl/output_file\\\"\\n                }\\n            ], \\n            \\\"steps\\\": [\\n                {\\n                    \\\"run\\\": \\\"#md5sum-tool.cwl\\\", \\n                    \\\"in\\\": [\\n                        {\\n                            \\\"source\\\": \\\"#md5sum-workflow.cwl/input_file\\\", \\n                            \\\"id\\\": \\\"#md5sum-workflow.cwl/md5sum/input_file\\\"\\n                        }\\n                    ], \\n                    \\\"out\\\": [\\n                        \\\"#md5sum-workflow.cwl/md5sum/output_file\\\"\\n                    ], \\n                    \\\"id\\\": \\\"#md5sum-workflow.cwl/md5sum\\\"\\n                }\\n            ], \\n            \\\"id\\\": \\\"#md5sum-workflow.cwl\\\"\\n        }\\n    ]\\n}\", \n",
      "  \"workflow_params\": {\n",
      "    \"input_file\": {\n",
      "      \"class\": \"File\", \n",
      "      \"location\": \"https://raw.githubusercontent.com/dockstore-testing/md5sum-checker/develop/md5sum.input\"\n",
      "    }, \n",
      "    \"expected_md5\": \"00579a00e3e7fa0674428ac7049423e2\"\n",
      "  }, \n",
      "  \"workflow_type_version\": \"v1.0\", \n",
      "  \"workflow_type\": \"CWL\"\n",
      "}\n",
      "INFO:synorchestrator.orchestrator:Submitting job '290529211034671585' for eval 'wflow3' to WES endpoint 'hca-cromwell'\n",
      "INFO:synorchestrator.orchestrator:Submitting job '290529211036592693' for eval 'wflow3' to WES endpoint 'broad-cromwell'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:synorchestrator.orchestrator:Submitting job '290529211040766509' for eval 'wflow2' to WES endpoint 'broad-cromwell'\n",
      "INFO:synorchestrator.orchestrator:Submitting job '290529211038681861' for eval 'wflow2' to WES endpoint 'hca-cromwell'\n",
      "INFO:synorchestrator.orchestrator:Submitting job '290529211048309644' for eval 'wflow0' to WES endpoint 'broad-cromwell'\n",
      "INFO:synorchestrator.orchestrator:Submitting job '290529211044953971' for eval 'wflow0' to WES endpoint 'arvados-wes'\n"
     ]
    }
   ],
   "source": [
    "submissions = orchestrator.run_all({\n",
    "    'wflow0': ['arvados-wes', 'broad-cromwell'],\n",
    "    'wflow2': ['hca-cromwell', 'broad-cromwell'],\n",
    "    'wflow3': ['hca-cromwell', 'broad-cromwell']\n",
    "})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>submission_status</th>\n",
       "      <th>elapsed_time</th>\n",
       "      <th>job</th>\n",
       "      <th>wes_id</th>\n",
       "      <th>queue_id</th>\n",
       "      <th>run_status</th>\n",
       "      <th>run_id</th>\n",
       "      <th>start_time</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">TopMed_Variant_Caller</th>\n",
       "      <th>290529211038681861</th>\n",
       "      <td>SUBMITTED</td>\n",
       "      <td>0h:1m:12s</td>\n",
       "      <td>checker</td>\n",
       "      <td>hca-cromwell</td>\n",
       "      <td>TopMed_Variant_Caller</td>\n",
       "      <td>COMPLETE</td>\n",
       "      <td>824d644b-f776-4392-aa96-2bd27016fbc4</td>\n",
       "      <td>Tue May 29 21:10:50 2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290529211040766509</th>\n",
       "      <td>SUBMITTED</td>\n",
       "      <td>0h:20m:20s</td>\n",
       "      <td>checker</td>\n",
       "      <td>broad-cromwell</td>\n",
       "      <td>TopMed_Variant_Caller</td>\n",
       "      <td>COMPLETE</td>\n",
       "      <td>c4732d4e-33df-4b8d-bed1-645dc8239af3</td>\n",
       "      <td>Tue May 29 21:10:50 2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">md5sum-checker</th>\n",
       "      <th>290529211044953971</th>\n",
       "      <td>SUBMITTED</td>\n",
       "      <td>0h:0m:32s</td>\n",
       "      <td>checker</td>\n",
       "      <td>arvados-wes</td>\n",
       "      <td>md5sum-checker</td>\n",
       "      <td>COMPLETE</td>\n",
       "      <td>qr1hi-xvhdp-as5qntn3l0eqfxq</td>\n",
       "      <td>Tue May 29 21:10:51 2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290529211048309644</th>\n",
       "      <td>SUBMITTED</td>\n",
       "      <td>0h:0m:22s</td>\n",
       "      <td>checker</td>\n",
       "      <td>broad-cromwell</td>\n",
       "      <td>md5sum-checker</td>\n",
       "      <td>EXECUTOR_ERROR</td>\n",
       "      <td>aa5eb354-93ec-4685-8bd2-4e9fb5f99dd4</td>\n",
       "      <td>Tue May 29 21:10:50 2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">u_of_Michigan_alignment_pipeline</th>\n",
       "      <th>290529211034671585</th>\n",
       "      <td>SUBMITTED</td>\n",
       "      <td>0h:1m:22s</td>\n",
       "      <td>checker</td>\n",
       "      <td>hca-cromwell</td>\n",
       "      <td>u_of_Michigan_alignment_pipeline</td>\n",
       "      <td>COMPLETE</td>\n",
       "      <td>3ab634d4-f952-4809-9913-e6948d573bbd</td>\n",
       "      <td>Tue May 29 21:10:49 2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290529211036592693</th>\n",
       "      <td>SUBMITTED</td>\n",
       "      <td>0h:27m:39s</td>\n",
       "      <td>checker</td>\n",
       "      <td>broad-cromwell</td>\n",
       "      <td>u_of_Michigan_alignment_pipeline</td>\n",
       "      <td>COMPLETE</td>\n",
       "      <td>9d1c69a3-952e-47bb-9f0e-ef9cc4d6f46f</td>\n",
       "      <td>Tue May 29 21:10:49 2018</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    submission_status elapsed_time      job  \\\n",
       "TopMed_Variant_Caller            290529211038681861         SUBMITTED    0h:1m:12s  checker   \n",
       "                                 290529211040766509         SUBMITTED   0h:20m:20s  checker   \n",
       "md5sum-checker                   290529211044953971         SUBMITTED    0h:0m:32s  checker   \n",
       "                                 290529211048309644         SUBMITTED    0h:0m:22s  checker   \n",
       "u_of_Michigan_alignment_pipeline 290529211034671585         SUBMITTED    0h:1m:22s  checker   \n",
       "                                 290529211036592693         SUBMITTED   0h:27m:39s  checker   \n",
       "\n",
       "                                                             wes_id  \\\n",
       "TopMed_Variant_Caller            290529211038681861    hca-cromwell   \n",
       "                                 290529211040766509  broad-cromwell   \n",
       "md5sum-checker                   290529211044953971     arvados-wes   \n",
       "                                 290529211048309644  broad-cromwell   \n",
       "u_of_Michigan_alignment_pipeline 290529211034671585    hca-cromwell   \n",
       "                                 290529211036592693  broad-cromwell   \n",
       "\n",
       "                                                                             queue_id  \\\n",
       "TopMed_Variant_Caller            290529211038681861             TopMed_Variant_Caller   \n",
       "                                 290529211040766509             TopMed_Variant_Caller   \n",
       "md5sum-checker                   290529211044953971                    md5sum-checker   \n",
       "                                 290529211048309644                    md5sum-checker   \n",
       "u_of_Michigan_alignment_pipeline 290529211034671585  u_of_Michigan_alignment_pipeline   \n",
       "                                 290529211036592693  u_of_Michigan_alignment_pipeline   \n",
       "\n",
       "                                                         run_status  \\\n",
       "TopMed_Variant_Caller            290529211038681861        COMPLETE   \n",
       "                                 290529211040766509        COMPLETE   \n",
       "md5sum-checker                   290529211044953971        COMPLETE   \n",
       "                                 290529211048309644  EXECUTOR_ERROR   \n",
       "u_of_Michigan_alignment_pipeline 290529211034671585        COMPLETE   \n",
       "                                 290529211036592693        COMPLETE   \n",
       "\n",
       "                                                                                   run_id  \\\n",
       "TopMed_Variant_Caller            290529211038681861  824d644b-f776-4392-aa96-2bd27016fbc4   \n",
       "                                 290529211040766509  c4732d4e-33df-4b8d-bed1-645dc8239af3   \n",
       "md5sum-checker                   290529211044953971           qr1hi-xvhdp-as5qntn3l0eqfxq   \n",
       "                                 290529211048309644  aa5eb354-93ec-4685-8bd2-4e9fb5f99dd4   \n",
       "u_of_Michigan_alignment_pipeline 290529211034671585  3ab634d4-f952-4809-9913-e6948d573bbd   \n",
       "                                 290529211036592693  9d1c69a3-952e-47bb-9f0e-ef9cc4d6f46f   \n",
       "\n",
       "                                                                   start_time  \n",
       "TopMed_Variant_Caller            290529211038681861  Tue May 29 21:10:50 2018  \n",
       "                                 290529211040766509  Tue May 29 21:10:50 2018  \n",
       "md5sum-checker                   290529211044953971  Tue May 29 21:10:51 2018  \n",
       "                                 290529211048309644  Tue May 29 21:10:50 2018  \n",
       "u_of_Michigan_alignment_pipeline 290529211034671585  Tue May 29 21:10:49 2018  \n",
       "                                 290529211036592693  Tue May 29 21:10:49 2018  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done\n"
     ]
    }
   ],
   "source": [
    "orchestrator.monitor(submissions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "from synorchestrator.wes.client import WESClient\n",
    "broadclient = WESClient(**config.wes_config['broad-cromwell'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "workflow_run = broadclient.get_workflow_run('aa5eb354-93ec-4685-8bd2-4e9fb5f99dd4')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{u'expected_md5': u'00579a00e3e7fa0674428ac7049423e2',\n",
       " u'input_file': {u'class': u'File',\n",
       "  u'location': u'https://raw.githubusercontent.com/dockstore-testing/md5sum-checker/develop/md5sum.input'}}"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "workflow_run['request']['workflow_params']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
